File: Solutions/Day15.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None 
class Solution: 
    def display(self,head):
        current = head
        while current:
            print(current.data,end=' ')
            current = current.next

    def insert(self,head,data): 
        if head==None:
            head = Node(data)
            return head
        else:
            temp = head
            while head.next!=None:
                head=head.next;
            newnode = Node(data)
            head.next = newnode
            return temp
    #Complete this method
mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)    
mylist.display(head); 	  

File: Solutions/Day15/Day15.go
package main

import(
	"fmt"
)

type Node struct {
	data int
	next *Node
}

type LinkedList struct {
	head *Node
}

func(list *LinkedList) Insert(item int) {
	newElement := &Node{data:item}
	if list.head == nil{
		list.head = newElement
	} else {
		start := list.head
		for list.head.next!=nil{
			list.head=list.head.next
		}
		list.head.next = newElement
		list.head=start
	}
}

func(list *LinkedList) Display(){
	start := list.head
	if list.head ==nil{
		return
	}
	for list.head!= nil{
			fmt.Printf("%d ",list.head.data)
			list.head=list.head.next
		}
	list.head = start

}


func NewList() *LinkedList {
	return &LinkedList{head:nil}
}

func main() {
	L := NewList()
	var t int
	var d int
	fmt.Scanf("%d\n",&t)
	for i:=0;i<t;i++{
		fmt.Scanf("%d\n",&d)
		L.Insert(d)
	}
	L.Display()
}
